### PR TITLE
✨(app-desk) display members

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

Display the members of a team in a datagrid.
Add dummy data for the moment.

## Proposal

- [x] Integrate design with DataGrid
- [x] Add endpoint `teams/${teamId}/accesses/?page=${page}`
- [x] Add dummy data for `teams/${teamId}/accesses/` endpoint
- [x] Test

## Demo

[scrnli_2_29_2024_4-03-16 PM.webm](https://github.com/numerique-gouv/people/assets/25994652/095316c9-7a72-45f8-8a06-03cb5f511f76)

### CODE REVIEW

File: src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            headerName: t('Name'),
+            field: 'user.name',
+          },
+          {
+            field: 'user.email',
+            headerName: t('Email'),
+          },
+          {
+            field: 'role',
+            headerName: t('Role'),
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx:80]: nit: based on the design this should be plural.

File: src/frontend/apps/e2e/__tests__/app-desk/menu.spec.ts [hunk 2]
```diff
@@ -58,19 +58,22 @@ test.describe('Menu', () => {
...
         await expect(
           page.getByRole('button', {
             name: 'Create a new team',
           }),
         ).toBeHidden();
 
         const reg = new RegExp(name.toLowerCase());
+        // eslint-disable-next-line playwright/no-conditional-expect
         await expect(page).toHaveURL(reg);
       }
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/menu.spec.ts:78]: nit: you could use ablock-comment to disable eslint (cf [here](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1)).

File: src/frontend/apps/desk/src/features/teams/api/types.tsx [hunk 3]
```diff
@@ -17,3 +17,8 @@ export interface Team {
   created_at: string;
   updated_at: string;
 }
+
+export interface User {
+  email: string;
+  name?: string;
+}
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/types.tsx:24]: should we store user id ?

File: src/frontend/apps/desk/src/features/teams/api/useTeamsAccesses.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import { PAGE_SIZE } from '../conf';
+
+import { dummyDataAPITeamAccesses } from './data/teams';
+import { Access } from './types';
+
+export type TeamAccessesAPIParams = {
+  page: number;
+  teamId: string;
+};
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useTeamsAccesses.tsx:13]: Question: do you plan to add sorting on columns on next PR?

File: src/frontend/apps/desk/src/features/teams/__tests__/PanelTeams.test.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import fetchMock from 'fetch-mock';
 
 import { AppWrapper } from '@/tests/utils';
 
-import { PanelTeams } from '../components/PanelTeams';
+import { PanelTeams } from '../components/Panel/TeamList';
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/__tests__/PanelTeams.test.tsx:7]: nit: `TeamsList` or  `Teams`  might be a better naming.

File: src/frontend/apps/desk/src/features/teams/api/data/teams.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  const count = total;
+  const roles = Object.values(Role);
+  for (let i = 0; i < pageSize; i++) {
+    accesses.push({
+      id: faker.string.uuid(),
+      user: {
+        email: faker.internet.email(),
+        name: faker.person.fullName(),
+      },
+      role: roles[Math.floor(Math.random() * roles.length)],
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/data/teams.ts:16]: nit: take a look at [here](https://fakerjs.dev/api/helpers#enumvalue), might be useful for next mocks

File: src/frontend/apps/desk/src/features/teams/api/data/teams.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    });
+  }
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/data/teams.ts:18]: nit:
```suggestion
export function createRandomAccess(): TeamAccess {
  return {
    id: faker.string.uuid(),
    user: {
      email: faker.internet.email(),
      name: faker.person.fullName(),
    },
    role: faker.helpers.enumValue(Roles),
  };
}

export const dummyDataAPITeamAccesses = (total: number, pageSize: number) => {
  return {
    total,
    next: null,
    previous: null,
    results: faker.helpers.multiple(createRandomAccess, { count: pageSize }),
  };
};

```

File: src/frontend/apps/desk/src/features/teams/__tests__/PanelTeams.test.tsx (hunk 5 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/__tests__/PanelTeams.test.tsx:7]: or if we suppose we are in the `team/panel` subdirectory, having a `List.tsx` and `Item.tsx` might be perfect

File: src/frontend/apps/desk/src/features/teams/components/Panel/Panel.tsx [hunk 8]
```diff
@@ -4,8 +4,8 @@ import { useTranslation } from 'react-i18next';
 import { Box, Text } from '@/components';
 import { useCunninghamTheme } from '@/cunningham';
 
-import { PanelActions } from './PanelActions';
-import { PanelTeams } from './PanelTeams';
+import { PanelActions } from './Actions';
+import { PanelTeams } from './TeamList';
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/Panel/Panel.tsx:8]: nit: should we use relative path here ?

File: src/frontend/apps/desk/src/features/teams/components/Panel/TeamList.tsx [hunk 9]
```diff
@@ -4,12 +4,10 @@ import { useTranslation } from 'react-i18next';
...
 import { InfiniteScroll } from '@/components/InfiniteScroll';
+import { Team, useTeams } from '@/features/teams/api/';
+import { useTeamStore } from '@/features/teams/store/useTeamsStore';
 
-import { Team } from '../api/types';
-import { useTeams } from '../api/useTeams';
-import { useTeamStore } from '../store/useTeamsStore';
-
-import { PanelTeam } from './PanelTeam';
+import { PanelTeam } from './TeamItem';
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/Panel/TeamList.tsx:10]: same

File: src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  const { data, isLoading, error } = useTeamAccesses({
+    teamId: team.id,
+    page,
+  });
+
+  const accesses = data?.results;
+
+  useEffect(() => {
+    setPagesCount(data?.count ? Math.ceil(data?.count / pageSize) : 0);
+  }, [data?.count, pageSize, setPagesCount]);
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx:32]: I might be wrong, suggestion, please feel free to ignore it.
Why is `setPagesCount` a dependency ?

```suggestion
  useEffect(() => {
    if (!accesses) {
      setPagesCount(0);
    }
    setPagesCount(Math.ceil(accesses.length / pageSize));
  }, [accesses, pageSize, page]);
```

**Review by lebaudantoine — COMMENTED** [review]
Nice one, I luv the interface, it will be so efficient to manage a group ❤️
thx for the extra effort while adapting the screen to the newly designed front.
Again, please feel free ignoring my nit-picking comment, I won't take it wrong

**Comment by lebaudantoine** [comment]
Might be great to translate the default message as well

Capture d’écran 2024-03-01 à 16 00 26

File: src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx (hunk 10 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/components/Member/MemberGrid.tsx]: I close it @AntoLC !

**Review by lebaudantoine — APPROVED** [review]
I approved it ! The only missing feature concerns the empty message for the data grid, which is not translated. Either push a fix up or open an issue, so we fix it later, as it's not the top prio.

**Comment by AntoLC** [comment]
> I approved it ! The only missing feature concerns the empty message for the data grid, which is not translated. Either push a fix up or open an issue, so we fix it later, as it's not the top prio.

Yes, we don't have the hand on it, I will open an issue on Cunningham.

**Pull request merged**